python/pyarrow/array.pxi

```diff
@@ -3104,12 +3109,18 @@ cdef class ExtensionArray(Array):
 
     def _to_pandas(self, options, **kwargs):
         pandas_dtype = None
-        try:
-            pandas_dtype = self.type.to_pandas_dtype()
-        except NotImplementedError:
-            pass
+        types_mapper = kwargs.get('types_mapper', None)
+
+        if types_mapper:
+            pandas_dtype = types_mapper(self.type)
+        elif issubclass(type(self.type), ExtensionType):
```

Comment:
This check isn't needed here, since this is in the ExtensionArray subclass. 

But more generally, I am wondering if this method is actually still needed now we are adding similar logic to the base class' `to_pandas`

Reply:
The only difference here now is calling `self.type.to_pandas_dtype()` vs `self.storage.type.to_pandas_dtype()` (storage gets passed to the base Array function).

Reply:
I did remove the issubclass check in a recent refactor!

Reply:
Currently we are indeed passing down `self.storage`, but I seem to remember that we added similar logic to the underlying C++ ConvertArrayToPandas to also (by default) use the storage array to do the actual conversion. So it _might_ actually work by just removing the overridden method completely.

Reply:
Looks like you are right! https://github.com/apache/arrow/blob/0cea12ffc87915e3808d153d4d8e5f115e48d487/python/pyarrow/src/arrow/python/arrow_to_pandas.cc#L2457

Running pytests locally seems to agree as well. I'll run the integration tests with this change to verify.